src/modules/Dropdown/Dropdown.js

```diff
@@ -328,13 +329,15 @@ export default class Dropdown extends Component {
     // prevent selecting null if there was no selected item value
     if (!value) return
 
-    this.setValue(value)
 
     // notify the onChange prop that the user is trying to change value
     if (multiple) {
       // state value may be undefined
-      this.onChange(e, _.union(this.state.value, [value]))
+      const newValue = _.union(this.state.value, [value])
+      this.setValue(newValue)
+      this.onChange(e, newValue)
```

Comment:
This should arguably be handled by trySetState, but we're handling in the component for now.